contributing/writing-css.md

```diff
@@ -0,0 +1,204 @@
+# Writing CSS For AMP Runtime?
+
+- [Who should read this doc?](#who-should-read-this-doc)
+- [What is Specificity?](#what-is-specificity)
+  - [Example CSS selectors](#example-css-selectors)
+  - [How to calculate Specificity](#how-to-calculate-specificity)
+- [Writing CSS in AMP Runtime or Extensions](#writing-css-in-amp-runtime-or-extensions)
+  - [Creating a new Selector:](#creating-a-new-selector)
+  - [Modifying an existing Selector:](#modifying-an-existing-selector)
+  - [FOUC](#fouc)
+  - [Rule of thumb](#rule-of-thumb)
+- [Case Studies](#case-studies)
+  - [AMP Selector CSS has high CSS specificity](#amp-selector-css-has-high-css-specificity)
+  - [AMP Accordion CSS Changes broke specificity](#amp-accordion-css-changes-broke-specificity)
+- [Reading material](#reading-material)
+
+## Who should read this doc?
+
+Are you a contributor to ampproject/amphtml, so you make changes to the runtime or add or
+modify amp-extensions? Then you should definitely read this document.
+
+Do you write solutions on top of the amp library (ampstart/ABE), this may be a good read for
+you.
+
+## What is Specificity?
+
+Specificity is the means by which browsers decide which CSS property values are the most
+relevant to an element and, therefore, will be applied. Specificity is based on the matching rules
+which are composed of different sorts of CSS selectors.
+
+
+A selector is something which can identify/select an element or a group of elements and apply a
+list of properties to all the selected elements
+
+### Example CSS selectors
+
+```css
+ *
+.favorite
+ul#summer-drinks li.favorite
+html body div#pagewrap ul#summer-drinks li.favorite
+html > body div#pagewrap ul#summer-drinks > li.favorite
+.ampstart-dropcap:first-letter
+#summer-drinks::before
+#summer-drinks::after
+```
+A selector can contain a class, id, pseudo elements, psuedo classes , :not(), and a combination
+of any of these and many more(Here is a comprehensive list of all the selectors -
+https://developer.mozilla.org/en-US/docs/Web/CSS/Reference#Selectors​).
+
+In general, the order of the CSS selectors do not affect which rules get applied to your CSS,
+unless the selectors have the same specificity and apply to at-least one element in common.
+A well written CSS will work perfectly well even when the selectors are re-ordered (example
+scenario- reordering imports in a css file). In real world it is hard to achieve this , especially
+when we import CSS via gulp or 3p css (which can change at any time)
+
+### How to calculate Specificity
+
+This blog explains specificty in very simple terms and helps to understand how it is computed -
+https://css-tricks.com/specifics-on-css-specificity/​ (very short and understandable - I promise)
```

Comment:
Clicking the "View" button, to see the compiled version of this markdown file, you will see that this link points to `https://css-tricks.com/specifics-on-css-specificity/%E2%80%8B` (and the site redirects you to another page)